src/os/ObjectStore.h

```diff
@@ -932,8 +933,10 @@ class ObjectStore {
       Op* decode_op() {
         assert(ops > 0);
 
+        op_buffer_p = op_bl_p.get_current_ptr().c_str();
```

Comment:
this is making a dangerous assumption that the buffer is contiguous for the size of Op.  This is get get_contiguous does above.. you would pass in sizeof(Op) here tho instead of t->data.ops * sizeof(Op).

I'm not convinced this is going to help much, though.  Instead it seems like we should aim to have the buffer large enoguh to accomodate the # of ops we expect when it is initially allocated (say, aim for 8 or whatever) and waste a bit of memory.  At least, we should do that too!